src/main/scala/firrtl/Emitter.scala

```diff
@@ -308,28 +304,37 @@ class VerilogEmitter extends Emitter {
         }
 
         def addUpdate(e: Expression, tabs: String): Seq[Seq[Any]] = {
-          netlist.getOrElse(e, e) match {
+          if (weq(e, r)) Nil else netlist getOrElse (e, e) match {
             case m: Mux if canFlatten(m) => {
-              val ifStatement = Seq(tabs, "if(", m.cond, ") begin")
+              val ifStatement = Seq(tabs, "if (", m.cond, ") begin")
               val trueCase = addUpdate(m.tval, tabs + tab)
               val elseStatement = Seq(tabs, "end else begin")
+              val ifNotStatement = Seq(tabs, "if (!(", m.cond, ")) begin")
               val falseCase = addUpdate(m.fval, tabs + tab)
               val endStatement = Seq(tabs, "end")
 
-              if (falseCase.isEmpty)
-                ifStatement +: trueCase :+ endStatement
-              else
-                ifStatement +: trueCase ++: elseStatement +: falseCase :+ endStatement
+              (trueCase.nonEmpty, falseCase.nonEmpty) match {
+                case (true, true) =>
+                  ifStatement +: trueCase ++: elseStatement +: falseCase :+ endStatement
+                case (true, false) =>
+                  ifStatement +: trueCase :+ endStatement
+                case (false, true) =>
+                  ifNotStatement +: falseCase :+ endStatement
+                case (false, false) =>
+                  error("Can't happen!")
```

Comment:
I think it's more consistent with the rest of this code to let it be a match error, rather than catching the bad case and explicitly erroring.